OpenRA.Game/GameRules/WeaponInfo.cs

```diff
@@ -64,6 +64,8 @@ public WarheadArgs(WarheadArgs args)
 		public WarheadArgs() { }
 	}
 
+	public enum BurstMode { Individual, AllOnInitialTargetPosition }
```

Comment:
How about `Fixed` as a less-of-a-mouthful name than `AllOnInitialTargetPosition`?

Reply:
Ok. I'm not entirely sure if `Static` alone would be enough though.

The other 2 modes I have in mind for a follow-up are
(1) follows target like `Individual`, but cancels burst (drops any remaining bursts to zero) once target becomes invalid in some way
(2) always depletes all bursts no matter what, but follows target instead of shooting at fixed position as long as target is alive and moving.

Suggestions on how to call these, to avoid having to rename the two introduced here later?

Would something like
`Flexible` (currently `Individual`)
`CancelOnTargetDeath` (1)
`DepleteOnTargetDeath` (2)
`Static`
be ok?
Not sure how to shorten the middle ones without making it hard to differentiate them...

Reply:
I've put less thought into this than you have, but given those use cases my first thought would to be use separate flags:
```csharp
[Desc("Update the firing position to follow targets if they move between bursts?")]
public readonly bool BurstTracking = true;

[Desc("Allow the weapon to cancel its target between bursts if requested by the player or the target dies?")]
public readonly bool BurstInterruptable = true;

[Desc("Reset the burst counter to 0 when switching targets?")]
public readonly bool BurstResetOnTargetChange = false;

```

It would be ideal if we could also fix `FireDelay` to update the target position if `BurstTracking = true` / `BurstMode = Individual`.